renderdoc/driver/gl/gl_renderstate.cpp

```diff
@@ -26,42 +26,83 @@
+    eGL_DEPTH_CLAMP,
+    eGL_DEPTH_TEST,
+    eGL_DEPTH_BOUNDS_TEST_EXT,
+    eGL_DITHER,
+    eGL_FRAMEBUFFER_SRGB,
+    eGL_LINE_SMOOTH,
+    eGL_MULTISAMPLE,
+    eGL_POLYGON_SMOOTH,
+    eGL_POLYGON_OFFSET_FILL,
+    eGL_POLYGON_OFFSET_LINE,
+    eGL_POLYGON_OFFSET_POINT,
+    eGL_PROGRAM_POINT_SIZE,
+    eGL_PRIMITIVE_RESTART,
+    eGL_PRIMITIVE_RESTART_FIXED_INDEX,
+    eGL_SAMPLE_ALPHA_TO_COVERAGE,
+    eGL_SAMPLE_ALPHA_TO_ONE,
+    eGL_SAMPLE_COVERAGE,
+    eGL_SAMPLE_MASK,
+    eGL_SAMPLE_SHADING,
+    eGL_RASTER_MULTISAMPLE_EXT,
+    eGL_STENCIL_TEST,
+    eGL_TEXTURE_CUBE_MAP_SEAMLESS,
+    eGL_BLEND_ADVANCED_COHERENT_KHR,
+    eGL_RASTERIZER_DISCARD,
+};
+
 void PixelUnpackState::Fetch(const GLHookSet *funcs, bool compressed)
 {
-  funcs->glGetIntegerv(eGL_UNPACK_SWAP_BYTES, &swapBytes);
+  if(!IsGLES)
+    funcs->glGetIntegerv(eGL_UNPACK_SWAP_BYTES, &swapBytes);
   funcs->glGetIntegerv(eGL_UNPACK_ROW_LENGTH, &rowlength);
   funcs->glGetIntegerv(eGL_UNPACK_IMAGE_HEIGHT, &imageheight);
   funcs->glGetIntegerv(eGL_UNPACK_SKIP_PIXELS, &skipPixels);
   funcs->glGetIntegerv(eGL_UNPACK_SKIP_ROWS, &skipRows);
   funcs->glGetIntegerv(eGL_UNPACK_SKIP_IMAGES, &skipImages);
   funcs->glGetIntegerv(eGL_UNPACK_ALIGNMENT, &alignment);
 
-  if(compressed)
-  {
-    funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_WIDTH, &compressedBlockWidth);
-    funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_HEIGHT, &compressedBlockHeight);
-    funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_DEPTH, &compressedBlockDepth);
-    funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_SIZE, &compressedBlockSize);
-  }
+  if(!IsGLES)
+    if(compressed)
+    {
+      funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_WIDTH, &compressedBlockWidth);
+      funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_HEIGHT, &compressedBlockHeight);
+      funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_DEPTH, &compressedBlockDepth);
+      funcs->glGetIntegerv(eGL_UNPACK_COMPRESSED_BLOCK_SIZE, &compressedBlockSize);
+    }
 }
 
 void PixelUnpackState::Apply(const GLHookSet *funcs, bool compressed)
 {
-  funcs->glPixelStorei(eGL_UNPACK_SWAP_BYTES, swapBytes);
+  if(!IsGLES)
+    funcs->glPixelStorei(eGL_UNPACK_SWAP_BYTES, swapBytes);
   funcs->glPixelStorei(eGL_UNPACK_ROW_LENGTH, rowlength);
   funcs->glPixelStorei(eGL_UNPACK_IMAGE_HEIGHT, imageheight);
   funcs->glPixelStorei(eGL_UNPACK_SKIP_PIXELS, skipPixels);
   funcs->glPixelStorei(eGL_UNPACK_SKIP_ROWS, skipRows);
   funcs->glPixelStorei(eGL_UNPACK_SKIP_IMAGES, skipImages);
   funcs->glPixelStorei(eGL_UNPACK_ALIGNMENT, alignment);
 
-  if(compressed)
-  {
-    funcs->glPixelStorei(eGL_UNPACK_COMPRESSED_BLOCK_WIDTH, compressedBlockWidth);
-    funcs->glPixelStorei(eGL_UNPACK_COMPRESSED_BLOCK_HEIGHT, compressedBlockHeight);
-    funcs->glPixelStorei(eGL_UNPACK_COMPRESSED_BLOCK_DEPTH, compressedBlockDepth);
-    funcs->glPixelStorei(eGL_UNPACK_COMPRESSED_BLOCK_SIZE, compressedBlockSize);
-  }
+  if(!IsGLES)
+    if(compressed)
```

Comment:
Can you either put {} around the outer if, or in this case just combine the conditions? I only like to have brace-less ifs for single line statements.